src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -710,7 +717,7 @@ class IntervalExpressionSegment(BaseSegment):
     [
         RegexLexer(
             "at_sign",
-            r"[@][a-zA-Z0-9_]*",
+            r"@@?[a-zA-Z0-9_.]*",
```

Comment:
I think `.` here means "match any character. Should this be `\.`?

As written, I think it will also allow the name to begin and/or end with a `.`. That worries me a little. Can we prevent this?

Reply:
I double checked, metacharacters (except `\`) are matched literally inside a character class in Python regex. This fails as expected on a variable with an invalid char in the name : `echo 'SELECT @var&broken | sqlfluff parse - --dialect mysql`.

For your second point, this indeed looks wrong, but I've found MySQL is so bizarre this is actually valid. The [documentation](https://dev.mysql.com/doc/refman/8.0/en/user-variables.html) states:

> User variables are written as @var_name, where the variable name var_name consists of alphanumeric characters, ., _, and $. A user variable name can contain other characters if you quote it as a string or identifier (for example, @'my-var', @"my-var", or @\`my-var\`).

I hardly believe anyone in their right mind would use a variable name with a `.` at the start or end, so I'm happy to change it to something like `r"@@?[a-zA-Z0-9_]+\.?[a-zA-Z0-9_]*"`.  The `$` sign should be added as well, just let me know what's preferable for the tool and I'll prepare the commit

Reply:
The above looks close. I think you want the dot and the remainder to look like ‘(\.[a-zA-Z0-9_]+)?`, to prevent it ending with a dot.

Reply:
Alright, I've pushed the change with the $ sign also supported, and added more fixtures for session variables. Thanks for the feedback!